lib/travis/build/addons/jwt.rb

```diff
@@ -0,0 +1,41 @@
+require 'travis/build/addons/base'
+require 'jwt'
+
+module Travis
+  module Build
+    class Addons
+      class Jwt < Base
+        SUPER_USER_SAFE = true
+
+        def before_before_script
+          secrets = nil
+          if config.is_a?(String)
+            secrets = [config]
+          else
+            secrets = config.values
+          end
+
+          tokens = {}
+          secrets.each do |secret|
+            key, secret = secret.split('=').map(&:strip)
+            pull_request = self.data.pull_request ? self.data.pull_request : ""
+            now = Time.now.to_i()
+            payload = {"iss" => "travis-ci.org",
+                      "slug" => self.data.slug,
+                      "pull-request" => pull_request,
+                      "exp" => now+5400,
+                      "iat" => now}
+            token = JWT.encode(payload, secret)
```

Comment:
Every way I look at it, JWT.encode doesn't throw an exception
I can certainly handle an exception in case, but the only way I figure is if payload isn't a valid object.

@BanzaiMan as you said earlier, config could be null? Any idea how to replicate that?
`let(:config) { nil; }` returns still makes config = `{}`